### PR TITLE
fix: validate priority fee cache env config

### LINKED ISSUE #293: Medium: Priority fee estimator accepts malformed cache env values disabling cache hits and cache cap
## Summary

`HeliusPriorityFeeEstimator` reads cache-related runtime configuration from environment variables using raw `parseInt(...)` and uses the parsed values directly.

Affected environment variables:

* `KEEPER_PRIORITY_FEE_CACHE_MS`
* `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES`

Malformed values such as `abc` become `NaN`. This can disable cache hits and bypass the intended cache size cap.

This is an operational reliability issue because the priority fee estimator can perform unnecessary repeated external RPC calls and allow the internal cache to grow beyond the intended maximum size.

## Severity

Medium.

This is not a direct fund-loss issue, but it can affect keeper reliability and operational cost:

* malformed cache duration disables cache hits,
* repeated calls fetch again instead of using the cached estimate,
* malformed cache max entries disables cache eviction,
* cache can grow beyond the intended cap,
* external priority fee estimation can be called more often than expected,
* long-running keepers may accumulate more memory usage than intended.

## Duplicate Check

I checked open issues and open pull requests using the following keywords:

* `KEEPER_PRIORITY_FEE_CACHE_MS`
* `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES`
* `priority fee cache env`
* `cacheMaxEntries`
* `NaN`
* `malformed`

No open issue or open pull request with the same root cause or same PoC was found at the time of reporting.

## Affected Code

In `src/lib/priority-fee.ts`, the estimator constructor parses cache env values directly:

```ts
this._cacheMs =
  opts?.cacheMs ??
  parseInt(process.env.KEEPER_PRIORITY_FEE_CACHE_MS ?? String(DEFAULT_CACHE_MS), 10);

this._cacheMaxEntries =
  opts?.cacheMaxEntries ??
  parseInt(
    process.env.KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES ?? String(DEFAULT_CACHE_MAX_ENTRIES),
    10,
  );
```

The defaults are:

```ts
const DEFAULT_CACHE_MS = 5_000;
const DEFAULT_CACHE_MAX_ENTRIES = 1_000;
```

However, malformed env values do not fall back to these defaults once the env variable is present.

## Root Cause

The root cause is that `parseInt(...)` can return `NaN`, but the parsed value is not validated before being assigned to internal cache configuration.

Example:

```ts
parseInt("abc", 10); // NaN
```

### Case 1: malformed cache duration

If the keeper is started with:

```env
KEEPER_PRIORITY_FEE_CACHE_MS=abc
```

then:

```ts
this._cacheMs = NaN;
```

When a cache entry is stored, the estimator computes:

```ts
expiresAt: Date.now() + this._cacheMs
```

This becomes:

```ts
expiresAt: NaN
```

Later, cache hits are checked with:

```ts
if (cached && Date.now() < cached.expiresAt) {
  return cached.value;
}
```

Since `Date.now() < NaN` is always false, the cache entry is never considered valid. The estimator fetches again even when the same account keys and tier are requested immediately afterward.

### Case 2: malformed cache max entries

If the keeper is started with:

```env
KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES=abc
```

then:

```ts
this._cacheMaxEntries = NaN;
```

Cache eviction is controlled by:

```ts
while (this._cache.size >= this._cacheMaxEntries) {
  ...
}
```

Since comparison against `NaN` is always false, the eviction loop never runs. The cache can grow beyond the intended default cap of `1000` entries.

## Proof of Concept

The following PoC tests were added to `tests/lib/priority-fee.test.ts` inside the existing `HeliusPriorityFeeEstimator` / `M-4: bounded cache size` test suite.

### PoC 1: malformed `KEEPER_PRIORITY_FEE_CACHE_MS` disables cache hits

```ts
it("PoC: malformed KEEPER_PRIORITY_FEE_CACHE_MS should fall back to default cache duration", async () => {
  const origEnv = process.env.KEEPER_PRIORITY_FEE_CACHE_MS;
  process.env.KEEPER_PRIORITY_FEE_CACHE_MS = "abc";

  try {
    const fetchFn = mockFetch(HELIUS_SUCCESS_RESPONSE);
    global.fetch = fetchFn;

    const estimator = new HeliusPriorityFeeEstimator("https://rpc.example.com");

    await estimator.estimate(["acc1"], "crank");
    await estimator.estimate(["acc1"], "crank");

    // Expected: malformed cache duration should fall back to DEFAULT_CACHE_MS,
    // so the second call should be served from cache.
    //
    // Current behavior: parseInt("abc", 10) becomes NaN, expiresAt becomes NaN,
    // Date.now() < NaN is false, and the second call fetches again.
    expect(fetchFn).toHaveBeenCalledTimes(1);
  } finally {
    if (origEnv === undefined) delete process.env.KEEPER_PRIORITY_FEE_CACHE_MS;
    else process.env.KEEPER_PRIORITY_FEE_CACHE_MS = origEnv;
  }
});
```

### PoC 2: malformed `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES` disables cache size cap

```ts
it("PoC: malformed KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES should fall back to default cache cap", async () => {
  const origEnv = process.env.KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES;
  process.env.KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES = "abc";

  try {
    const fetchFn = mockFetch(HELIUS_SUCCESS_RESPONSE);
    global.fetch = fetchFn;

    const estimator = new HeliusPriorityFeeEstimator("https://rpc.example.com", {
      cacheMs: 60_000,
    });

    for (let i = 0; i < 1_005; i++) {
      await estimator.estimate([`acc${i}`], "crank");
    }

    // Expected: malformed max entries should fall back to DEFAULT_CACHE_MAX_ENTRIES = 1000.
    //
    // Current behavior: parseInt("abc", 10) becomes NaN,
    // size >= NaN is false, and the eviction loop never runs.
    expect(cacheSize(estimator)).toBeLessThanOrEqual(1_000);
  } finally {
    if (origEnv === undefined) delete process.env.KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES;
    else process.env.KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES = origEnv;
  }
});
```

## Test Command

```bash
pnpm test -- tests/lib/priority-fee.test.ts
```

## Test Result

The PoC fails as expected and demonstrates the bug.

### Failure 1: malformed cache duration disables cache hit

```text
FAIL tests/lib/priority-fee.test.ts > HeliusPriorityFeeEstimator > M-4: bounded cache size > PoC: malformed KEEPER_PRIORITY_FEE_CACHE_MS should fall back to default cache duration

AssertionError: expected "vi.fn()" to be called 1 times, but got 2 times
```

This proves that the second call was not served from cache. Instead, the estimator fetched again.

Expected behavior:

```text
fetch called 1 time
```

Actual behavior:

```text
fetch called 2 times
```

### Failure 2: malformed cache max entries disables cache cap

```text
FAIL tests/lib/priority-fee.test.ts > HeliusPriorityFeeEstimator > M-4: bounded cache size > PoC: malformed KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES should fall back to default cache cap

AssertionError: expected 1005 to be less than or equal to 1000
```

This proves that the cache exceeded the intended default max size of `1000` entries.

Expected behavior:

```text
cache size <= 1000
```

Actual behavior:

```text
cache size = 1005
```

### Final test summary

```text
Test Files  1 failed | 82 passed | 2 skipped
Tests       2 failed | 905 passed | 33 skipped
```

## Expected Behavior

Malformed or out-of-range env values should not silently become active runtime cache configuration.

Expected safe behavior:

* `KEEPER_PRIORITY_FEE_CACHE_MS=abc` should fall back to `DEFAULT_CACHE_MS`.
* `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES=abc` should fall back to `DEFAULT_CACHE_MAX_ENTRIES`.
* negative, zero, non-integer, or malformed cache values should not disable cache behavior.
* valid configured integer values should continue to work.

## Suggested Fix

Replace raw `parseInt(...)` usage with a bounded integer parser.

Example:

```ts
function parseBoundedIntEnv(
  name: string,
  fallback: number,
  min: number,
  max = Number.MAX_SAFE_INTEGER,
): number {
  const raw = process.env[name]?.trim();
  if (!raw) return fallback;

  const value = Number(raw);

  if (!Number.isInteger(value) || value < min || value > max) {
    return fallback;
  }

  return value;
}
```

Then use it for the cache config:

```ts
this._cacheMs =
  opts?.cacheMs ??
  parseBoundedIntEnv("KEEPER_PRIORITY_FEE_CACHE_MS", DEFAULT_CACHE_MS, 1);

this._cacheMaxEntries =
  opts?.cacheMaxEntries ??
  parseBoundedIntEnv(
    "KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES",
    DEFAULT_CACHE_MAX_ENTRIES,
    1,
  );
```

This preserves valid explicit config while preventing malformed values from disabling cache behavior.

### PR DESCRIPTION
**Pull request by Bayyan16**
## Summary

Validates priority fee estimator cache environment configuration before using it at runtime.

Previously, `KEEPER_PRIORITY_FEE_CACHE_MS` and `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES` were parsed with raw `parseInt(...)`. If either environment value was malformed, for example `abc`, the parsed result became `NaN` and was used directly by the cache logic.

This could disable cache hits or bypass the intended cache size cap.

Fixes #293

## Related Issue

This PR fixes #293.

Issue #293 reported that malformed priority fee cache environment values could become `NaN` and affect cache behavior:

* `KEEPER_PRIORITY_FEE_CACHE_MS=abc` could disable cache hits because cache entries were stored with `expiresAt = NaN`.
* `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES=abc` could disable cache eviction because the cache size comparison against `NaN` never passed.

## Changes

* Add bounded integer parsing for priority fee cache environment values.
* Fall back to `DEFAULT_CACHE_MS` when `KEEPER_PRIORITY_FEE_CACHE_MS` is missing, empty, malformed, non-integer, below the allowed minimum, or above the allowed maximum.
* Fall back to `DEFAULT_CACHE_MAX_ENTRIES` when `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES` is missing, empty, malformed, non-integer, below the allowed minimum, or above the allowed maximum.
* Preserve valid explicit integer configuration values.
* Add regression tests for malformed cache duration config.
* Add regression tests for malformed cache max entries config.

## Why

The priority fee estimator uses an in-memory cache to avoid repeatedly fetching the same priority fee estimate for the same account set and tier.

Before this change, malformed cache environment values could break that behavior.

### Malformed cache duration

If the keeper was configured with:

```env
KEEPER_PRIORITY_FEE_CACHE_MS=abc
```

then `parseInt("abc", 10)` returned `NaN`.

That caused cache entries to be stored with:

```ts
expiresAt = Date.now() + NaN
```

which produced:

```ts
expiresAt = NaN
```

Later, cache hits are checked with:

```ts
Date.now() < cached.expiresAt
```

Since comparison with `NaN` is always false, the cache entry was never considered valid. As a result, repeated calls for the same account set and tier fetched again instead of using the cached estimate.

### Malformed cache max entries

If the keeper was configured with:

```env
KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES=abc
```

then `parseInt("abc", 10)` returned `NaN`.

That made the cache eviction condition ineffective:

```ts
cache.size >= cacheMaxEntries
```

Since comparison with `NaN` is always false, the eviction loop did not run. This allowed the cache to grow beyond the intended default cap.

## Implementation

This PR adds a small bounded parser:

```ts
function parseBoundedIntEnv(
  name: string,
  fallback: number,
  min: number,
  max = Number.MAX_SAFE_INTEGER,
): number
```

The parser returns the fallback value when:

* the environment value is missing,
* the environment value is empty,
* the environment value is not an integer,
* the environment value is lower than the allowed minimum,
* the environment value is higher than the allowed maximum.

The parser is then used for:

```ts
KEEPER_PRIORITY_FEE_CACHE_MS
KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES
```

This keeps valid explicit configuration working while preventing malformed values from becoming active runtime cache config.

## Tests

Added regression coverage for:

* malformed `KEEPER_PRIORITY_FEE_CACHE_MS` falling back to the default cache duration,
* malformed `KEEPER_PRIORITY_FEE_CACHE_MAX_ENTRIES` falling back to the default cache cap.

Validated with:

```bash
pnpm test -- tests/lib/priority-fee.test.ts
pnpm build
pnpm test
```

Full test result:

```text
Test Files  84 passed | 2 skipped
Tests       907 passed | 33 skipped
```

### CODE REVIEW

File: src/lib/priority-fee.ts [hunk 1]
```diff
@@ -93,12 +109,13 @@ export class HeliusPriorityFeeEstimator implements PriorityFeeEstimator {
       "";
     this._cacheMs =
       opts?.cacheMs ??
-      parseInt(process.env.KEEPER_PRIORITY_FEE_CACHE_MS ?? String(DEFAULT_CACHE_MS), 10);
+      parseBoundedIntEnv("KEEPER_PRIORITY_FEE_CACHE_MS", DEFAULT_CACHE_MS, 1);
```
[the remark this answers is not in the document]
Comment (Bayyan16) [reply src/lib/priority-fee.ts]: Good catch. I updated the parser to allow `KEEPER_PRIORITY_FEE_CACHE_MS=0`, so zero TTL remains a valid explicit env override while malformed values still fall back to the default. I also added a regression test covering the zero TTL behavior.
